lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -973,6 +980,16 @@ NormalProtocolConformance *GetDistributedActorImplicitCodableRequest::evaluate(
     return nullptr;
   }
 
+  auto systemTy = getDistributedActorSystemType(nominal);
+  if (!systemTy || !systemTy->getAnyNominal()) {
+    return nullptr;
+  }
+
+  auto idTy = getDistributedActorIDType(nominal);
```

Comment:
Just wanted to mention it here but it could be done separately - we need to straighten up the invariants of result type for all the methods such as `getConcreteReplacementForProtocolActorSystemType`, `getDistributedActorSystemType`, and `getDistributedActorIDType` to make sure that they produce e.g. a null type (or `ErrorType`) if there was any kind of error or a (valid) system type. This way callers would just need a not null check.

Reply:
Yeah makes sense, I'll see what looks better and clean it up.